### PR TITLE
Allow cpu and gpu in int4wo and int4wo-gptq quantizer

### PR DESCRIPTION
**Pull request by jerryzh168**
Summary:
att

Test Plan:
verified in torchat

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by cpuhrsch** [comment]
Do we need to release a 0.1.1 for this?

**Review by cpuhrsch — APPROVED** [review]

**Comment by jerryzh168** [comment]
> Do we need to release a 0.1.1 for this?

it's fine, this is for torchat, and it will be using torchao-nightly. I'll still looking at some perf issue for this, I'll merge after that

File: torchao/quantization/GPTQ.py [hunk 1]
```diff
@@ -762,11 +762,15 @@ def _check_linear_int4_k(k, groupsize = 1, inner_k_tiles = None):
     return k_divisible_by_groupsize
 
 def linear_forward_int4(x, weight_int4pack, scales_and_zeros, out_features, groupsize):
```
Comment (cpuhrsch) [inline torchao/quantization/GPTQ.py:764]: So these conversions to bfloat16 are primarily needed because of `_weight_int4pack_mm`?

Comment (jerryzh168) [reply torchao/quantization/GPTQ.py:764]: yes, that's correct

**Comment by jerryzh168** [comment]
looks like there is no perf issues, I'll just merge

**Pull request merged**